docs/configuration_files.rst

```diff
@@ -0,0 +1,350 @@
+            low: 6.9
+            high: 7.1
+
+* `Reference a column`: columns in the pipeline can be referenced by their full name tagged with a dollar sign ``$``.
+  Example: the radius of cosmic voids seem to follow a lognormal distribution. You can create a table ``cosmic_voids``
+  with a column ``radii`` where you sample 10000 void sizes and two other columns, ``mean`` and ``variance`` that reference
+  the first column
+
+  .. code:: yaml
+
+    tables:
+      cosmic_voids:
+        radii: !scipy.stats.lognorm.rvs
+          s: 200.
+          size: 10000
+        mean: !numpy.mean
+          a: $cosmic_voids.radii
+        variance: !numpy.var
+          a: $cosmic_voids.radii
+
+
+* `Multi-column assignment`: if a function returns multiple columns, you can chose to assign them to multiple columns with different names or to a muti-column object.
+
+  Imagine you want the distribution of the circular velocities of 1000 halos following a Maxwellian distribution.
+  The histogram NumPy_ returns a 2-dimensional object. Therefore, you can choose
+
+  .. code:: yaml
+
+    tables:
+      halos:
+        circular_velocities: !scipy.stats.maxwell.rvs
+          s: 250
+          size: 1000
+        hist, bin_edges: !numpy.histogram
+          a: $halos.circular_velocities
+          density: True
+
+  or a multi-column assignment
+
+  .. code:: yaml
+
+    tables:
+      halos:
+        circular_velocities: !scipy.stats.maxwell.rvs
+          s: 250
+          size: 1000
+        histogram: !numpy.histogram
+          a: $halos.circular_velocities
+          density: True
+
+
+* `Referencing tables: table.init and table.complete dependencies`.
+
+  There are times when your function depends on tables. In these
+  cases, you need to ensure the referenced table has the necessary content and is not empty.
+
+  Example: you want to perform a very simple abundance matching, i.e. painting galaxies within your halos.
+  You can create two tables ``halos`` and ``galaxies`` storing the halo mass and galaxy luminosities.
+  Then you can stack these two tables and store it in a third table called ``matching``.
+
+  `Beware`: referencing tables is a bit different to variables or columns
+  -- where you simply tag a dollar sign.
+
+  The challenge: let ``tableA`` be a table with columns ``c1`` and ``c2``.
+  In configuration language, ``tableA`` is the name of the job.
+  *That means, when executing the configuration file, the first thing that happens is call ``tableA``, second,  call ``tableA.c1`` and third, call ``tableA.c2``.*
+  If you used the dollar sign to reference your ``tableA`` inside a function, this function might be called before the job ``tableA`` is complete, and the table will be empty.
+
+  `The solution`: to correctly reference tables, initialise your table with ``.init``, specify their dependences with the keyword ``.depends``
+  and ensure the tables are completed before calling the function with ``.complete``. Our example:
```

Comment:
This example if overly complicated; it is introducing multiple new features and conflating them when they are in fact separate:
- `.depends` can be used to force any function call to wait on the completion of any other job (probably should be mentioned as a bullet point in the "functions" section)
- Every table has an implicit `table_name.complete` job that occurs after all of its columns have been generated but does nothing
- By default tables are initialised using `astropy.table.Table()` however this can be overridden using the `.init` keyword to initialise the table with any function call